### PR TITLE
Add samples for IntentService

### PR DESCRIPTION
**Pull request by leenasn**
A sample on how to test IntentService with Robolectric, tests SharedPreference too

### CODE REVIEW

**Comment by leenasn** [comment]
Hi there,

Any update on when this pull request will be accepted? Thanks.

File: android-api-16/src/test/java/com/example/receiver/AlarmManagerReceiverTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.robolectric.Shadows.shadowOf;
+
+//@RunWith(RobolectricTestRunner.class)
+//@Config(constants = BuildConfig.class)
+// TODO: find a solution for the Issue: https://github.com/robolectric/robolectric/issues/2148 or another alternate for the same
+/*public class AlarmManagerReceiverTest {
+
+//    @Test
```
Comment (jongerrish) [inline android-api-16/src/test/java/com/example/receiver/AlarmManagerReceiverTest.java]: Commented out code?

File: android-api-19/src/test/java/com/example/service/SampleIntentServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Test
+    public void addsDataToSharedPreference(){
+        RoboSharedPreferences preferences = (RoboSharedPreferences) RuntimeEnvironment.application
+                .getSharedPreferences("example", Context.MODE_PRIVATE);
+        Intent intent =  new Intent(RuntimeEnvironment.application,SampleIntentService.class);
+        SampleIntentService registrationService = new SampleIntentService();
+        registrationService.onHandleIntent(intent);
+        assertNotSame(preferences.getString("SAMPLE_DATA", ""), "");
+
```
Comment (jongerrish) [inline android-api-19/src/test/java/com/example/service/SampleIntentServiceTest.java]: nit: can you be consistent with trailing blank lines in methods / class blocks.

File: android-api-21/src/main/java/com/example/receiver/BootCompleteReceiver.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.example.receiver;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+import com.example.service.SampleIntentService;
+
+public class BootCompleteReceiver extends BroadcastReceiver {
+    public BootCompleteReceiver() {
```
Comment (jongerrish) [inline android-api-21/src/main/java/com/example/receiver/BootCompleteReceiver.java]: Remove redundant default constructors, apply throughout

File: android-api-21/src/test/java/com/example/receiver/BootCompleteReceiverTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import static junit.framework.Assert.assertTrue;
+import static org.junit.Assert.*;
+
+@RunWith(RobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class)
+public class BootCompleteReceiverTest {
+
+    @Before
+    public void setUp() throws Exception {
+        ShadowLog.stream = System.out;
```
Comment (jongerrish) [inline android-api-21/src/test/java/com/example/receiver/BootCompleteReceiverTest.java]: What is this for? MAybe remove if its just for debugging, no other tests do this

**Comment by leenasn** [comment]
@jongerrish I've taken care of the feedback you had given. Can you take a look now?

**Pull request merged**